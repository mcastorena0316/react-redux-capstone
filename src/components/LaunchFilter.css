@import url('https://fonts.googleapis.com/css2?family=Muli:wght@500&display=swap');

.filter-div {
    text-align:center;
    display: flex;
    justify-content: space-around;
    font-family: 'Muli', sans-serif;
    font-size: 1.2em;
    align-items: center;
    margin: 14px 3px
}

.filter-div .filter-by {
    font-weight: bold;
    font-size: 1.2em;
}

.filter-div div select {
    height: 33px;
    border-radius: 5px;
    font-size: 0.9em;
    background-color:#dcf5fc;
}

.filter-div .location-div  select {
    max-width: 12vw;
    overflow: hidden
}

.year-div select {
    min-width: 60px;
}

@media only screen and (max-width: 600px) {

    .filter-div {
        flex-direction: column;
    }

    .filter-div div {
        width:84%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        
    }

    .filter-div .location-div select {
        max-width:20vw;
    }

    .filter-div div select {
        width: 20vw;
        height: 26px;
        margin: 5px;
    }
}

