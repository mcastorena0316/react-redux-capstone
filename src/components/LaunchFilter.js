
import React from 'react';
import PropTypes from 'prop-types';


const sucess = ['All', 'Yes', 'No', 'Pending'];

const LaunchFilter = ({ filter = 'All', handleChange }) => (
  <div>

    <span>LAUNCH SUCESS: </span>
    <select
      id="sucess"
      onChange={handleChange}
      value={filter}
    >
      {sucess.map(x => (
        <option
          key={Math.random()}
          value={x}
        >
          {x}
        </option>
      ))}
    </select>
  </div>
);


LaunchFilter.propTypes = {
  handleChange: PropTypes.func,

  filter: PropTypes.string,
};

LaunchFilter.defaultProps = {
  handleChange: () => { },
  filter: '',
};


export default LaunchFilter;
