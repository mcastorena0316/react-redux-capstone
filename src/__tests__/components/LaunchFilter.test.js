import React from 'react';
import { shallow } from 'enzyme';
// import toJson from 'enzyme-to-json';
// import configureStore from 'redux-mock-store';
import ReactDOM from 'react-dom';
import LaunchFilter from '../../components/LaunchFilter';

describe('LaunchFilter', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<LaunchFilter />);
    expect(component).toMatchSnapshot();
  });
});

describe('Test that 3 selects renders', () => {
  test('renders filter by launch sucess', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LaunchFilter />, div);
    expect(div.querySelector('#sucess').type).toBe('select-one');
    expect(div.querySelector('#sucess').textContent).toBe('AllYesNoPending');
  });
  test('renders filter by year success', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LaunchFilter />, div);
    expect(div.querySelector('#Date').type).toBe('select-one');
    expect(div.querySelector('#Date').textContent).toBe('All20202019201820172016201520142013201220102009200820072006');
  });

  test('renders filter by launch site', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LaunchFilter />, div);
    expect(div.querySelector('#Location').type).toBe('select-one');
  });
});
