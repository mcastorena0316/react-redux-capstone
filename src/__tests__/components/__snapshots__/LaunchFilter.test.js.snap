// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchFilter should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
